src/neuroconv/tools/testing/mock_ttl_signals.py

```diff
@@ -0,0 +1,290 @@
+        )
+        assert np.issubdtype(type(signal_mean), np.floating), (
+            "If specifying the 'signal_mean' manually, please ensure it matches the 'dtype'! "
+            f"Received '{type(signal_mean).__name__}', should be a float."
+        )
+        assert np.issubdtype(type(channel_noise), np.floating), (
+            "If specifying the 'channel_noise' manually, please ensure it matches the 'dtype'! "
+            f"Received '{type(channel_noise).__name__}', should be a float."
+        )
+
+    np.random.seed(seed=random_seed)
+    num_frames = np.ceil(signal_duration * sampling_frequency_hz).astype(int)
+    trace = (np.random.randn(num_frames) * channel_noise + baseline_mean).astype(dtype)
+
+    if ttl_times is not None:
+        ttl_times = np.array(ttl_times)
+    else:
+        # Start halfway through an off pulse
+        total_cycle_duration = ttl_off_duration + ttl_on_duration
+        cycle_start_times = np.arange(0, signal_duration, total_cycle_duration)
+        num_cycles = len(cycle_start_times)
+        num_ttl_pulses = (
+            num_cycles if signal_duration - cycle_start_times[-1] - ttl_off_duration > 0 else num_cycles - 1
+        )
+        ttl_times = np.linspace(start=ttl_off_duration / 2, stop=signal_duration, num=num_ttl_pulses)
+
+    ttl_start_frames = np.round(ttl_times * sampling_frequency_hz).astype(int)
+    num_frames_ttl_on_duration = np.round(ttl_on_duration * sampling_frequency_hz).astype(int)
+    ttl_on_durations = (slice(start, start + num_frames_ttl_on_duration) for start in ttl_start_frames)
+
+    for ttl_frames in ttl_on_durations:
+        trace[ttl_frames] += signal_mean
+
+    return trace
+
+
+def regenerate_test_cases(folder_path: FolderPathType, regenerate_reference_images: bool = False):
+    """
+    Regenerate the test cases of the file included in the main testing suite, which is frozen between breaking changes.
+
+    Parameters
+    ----------
+    folder_path: PathType
+        Folder to save the resulting NWB file in. For use in the testing suite, this must be the
+        '/test_testing/test_mock_ttl/' subfolder adjacent to the 'test_mock_tt.py' file.
+    regenerate_reference_images: boolean
+        If true, uses the kaleido package with plotly (you may need to install both) to regenerate the images used
+        as references in the documentation.
+    """
+    folder_path = Path(folder_path)
+
+    if regenerate_reference_images:
+        assert is_module_installed("kaleido") and is_module_installed(
+            "kaleido"
+        ), "To regenerate the reference images, you must install both plotly and kaleido!"
+        import plotly.graph_objects as go
+        from plotly.subplots import make_subplots
+
+        image_file_path = folder_path / "example_ttl_reference.png"
+
+    nwbfile_path = folder_path / "mock_ttl_examples.nwb"
+    compression_options = dict(compression="gzip", compression_opts=9)
```

Comment:
```suggestion
    compression_options = dict(compression="gzip", compression_opts=4)
```

why ? that seems needlessly aggressive

Reply:
Saves an extra 10% space on the ground truth data file, doesn't slow down the tests hardly at all. Regeneration of ground truth data is also near-instantaneous anyway.